File: tests/e2e-vm-jdk/src/main/java/org/jobrunr/tests/e2e/services/TestService.java
package org.jobrunr.tests.e2e.services;

import org.jobrunr.jobs.context.JobContext;
import org.jobrunr.scheduling.BackgroundJob;

import java.util.UUID;

public class TestService {

    public void doWork() {
        System.out.println("This is a test service");
    }

    public void doWork(Work work) {
        System.out.println("This is a test service " + work.getSomeId() + "; " + work.getSomeString() + "; " + work.getSomeInt() + "; " + work.getSomeLong());
    }

    public void doWorkWithMultipleParameters(JobContext jobContext, long someLongValue, Work work) {
        System.out.println("This is a test service with an extra long value: " + someLongValue + "; Work: " + work.getSomeId() + "; " + work.getSomeString() + "; " + work.getSomeInt() + "; " + work.getSomeLong());
    }

    public void run() {
        Work work = new Work(123, "some string", 456L, UUID.randomUUID());
        long someLongValue = 789L;

        BackgroundJob.enqueue(() -> this.doWorkWithMultipleParameters(JobContext.Null, someLongValue, work));
    }

}

File: language-support/jobrunr-kotlin-17-support/src/test/kotlin/org/jobrunr/scheduling/TestService.kt
package org.jobrunr.scheduling

import org.jobrunr.jobs.annotations.Job
import java.util.*

class TestService {

    fun doWork(s: String) {
        println(s)
    }

    fun doWorkWithDefaultParameter(id: UUID = UUID.randomUUID()) {
        println("id: $id")
    }

    @Job(name = "Some neat Job Display Name")
    fun doWorkWithJobName(s: String, jobName: String) {
        println("$s $jobName")
    }

}